### PR TITLE
Implemented Graphql for articles and comments and updated tests.

### PR DESCRIPTION
**Pull request by paulbrighton**

### CODE REVIEW

File: components/Comment.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { MarkdownPreview } from 'react-marked-markdown'
+
+class Comment extends Component {
+  render () {
+    const { comment } = this.props
+
+    return (
+      <div className='c-comment'>
+        <div className='c-comment__author'>
+          <img className='c-comment__img' src='../static/images/ruby.jpg' alt='A Ruby logo' />
```
Comment (krisquigley) [inline components/Comment.js]: Should the src here be their avatar, or is it default for everyone?

File: components/UserComments.js [hunk 2]
```diff
@@ -13,8 +16,10 @@ class UserComments extends Component {
         <a className='c-user-comments__login-btn' href='/auth/logout'>Logout</a>
       </div>
       <CommentForm />
-      <h4 className='c-user-comments__counter'>Add Comment Counter Here</h4 >
-      <Comments />
+      <h4 className='c-user-comments__counter'><Pluralize singular={'comment'} plural={'comments'} count={comments.length} /></h4>
+      {comments.map(comment => {
+        return <Comment comment={comment} key={comments.id} />
```
Comment (krisquigley) [inline components/UserComments.js]: key should be `comment.id`

File: test/components/featuredArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import FeaturedArticle from '../../components/FeaturedArticle'
+
+describe('<FeaturedArticle />', () => {
+  it('it renders correctly', () => {
+    const component = shallow(
+      <FeaturedArticle />
```
Comment (krisquigley) [inline test/components/featuredArticle.test.js]: I would still pass it required props, as tests are a form of documentation

File: test/components/latestArticles.test.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import LatestArticles from '../../components/FeaturedArticle'
+
+describe('<LatestArticles />', () => {
+  it('it renders correctly', () => {
+    const component = shallow(
+      <LatestArticles />
```
Comment (krisquigley) [inline test/components/latestArticles.test.js]: Pass in the required props

File: test/components/userComments.test.js [hunk 5]
```diff
@@ -2,8 +2,9 @@ import UserComments from '../../components/UserComments'
 
 describe('<UserComments  />', () => {
   it('it renders correctly', () => {
+    const comments = []
```
Comment (krisquigley) [inline test/components/userComments.test.js]: Pass in at least one comment

File: test/components/comment.test.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import Comment from '../../components/Comment'
+
+describe('<Comment />', () => {
+  it('it renders correctly', () => {
+    const comment = {
+      authorName: 'authorName'
```
Comment (krisquigley) [inline test/components/comment.test.js]: Pass in a full comment rather than just the authorName

File: test/components/Article.test.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import Article from '../../components/Article'
+
+describe('<Article />', () => {
+  it('it renders correctly', () => {
+    const article = {
```
Comment (krisquigley) [inline test/components/Article.test.js]: Check this against your graphql query to ensure it has all required keys

File: pages/article.js [hunk 8]
```diff
@@ -1,28 +1,49 @@
...
+        article(slug:$slug) {
+          author
+          title
+          text
+          createdAt
+          image
+          comments {
+            body
+            authorName
+            articleId
```
Comment (krisquigley) [inline pages/article.js]: Is this needed?

File: pages/article.js [hunk 9]
```diff
@@ -1,28 +1,49 @@
...
+            id
```
Comment (krisquigley) [inline pages/article.js]: Is this needed?

File: test/components/Article.test.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+      'author': 'Paul Brighton',
+      'title': 'New Article #3',
+      'text': 'A third article!!',
+      'createdAt': '2018-12-12 10:20:46 UTC',
+      'id': '3',
```
Comment (krisquigley) [inline test/components/Article.test.js]: Judging by your query, you aren't asking for an `id`

File: pages/article.js [hunk 11]
```diff
@@ -1,28 +1,51 @@
...
+            created_at
```
Comment (krisquigley) [inline pages/article.js]: I believe this should be `createdAt`

File: components/Article.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+class Article extends Component {
+  render () {
+    const { article } = this.props
+
+    return <article className='o-main-container'>
+      <div className='c-article'>
+        <h2 className='c-article__header'>{article.title}</h2>
+        <div className='c-header-divider' />
+        <h3 className='c-article__author'>Article by: {article.author}</h3>
+        <time className='c-article__published'><p>Published on: {moment(article.created_at).format('ll')}</p></time>
```
Comment (krisquigley) [inline components/Article.js]: this should be `createdAt`

File: components/Comment.js [hunk 13]
```diff
@@ -10,7 +10,7 @@ class Comment extends Component {
     return (
       <div className='c-comment'>
         <div className='c-comment__author'>
-          <img className='c-comment__img' src='../static/images/ruby.jpg' alt='A Ruby logo' />
+          <img className='c-comment__img' src={comment.image} />
           <h4 className='c-comment__author-name'>{comment.authorName}</h4>
           <time className='c-comment__created-date'>Created on: {moment(comment.created_at).format('ll')}</time>
```
[the remark this answers is not in the document]
Comment (krisquigley) [reply components/Comment.js]: This should be `createdAt`

File: pages/article.js [hunk 14]
```diff
@@ -1,28 +1,49 @@
...
+        article(slug:$slug) {
+          author
+          title
+          text
+          createdAt
+          image
+          comments {
+            body
+            authorName
+            CreatedAt
```
Comment (krisquigley) [inline pages/article.js]: should be lower case `c`. You should be able to confirm these queries by testing it through graphiql or insomnia

File: test/components/Article.test.js (hunk 10 above)
Comment (krisquigley) [reply test/components/Article.test.js]: `id` can be removed

File: test/components/comment.test.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import Comment from '../../components/Comment'
+
+describe('<Comment />', () => {
+  it('it renders correctly', () => {
+    const comment = {
+      authorName: 'authorName',
+      body: 'This is a comment',
+      image: '/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg'
```
Comment (krisquigley) [inline test/components/comment.test.js]: need to add a `createdAt` prop too

File: test/components/featuredArticle.test.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
+import FeaturedArticle from '../../components/FeaturedArticle'
+
+describe('<FeaturedArticle />', () => {
+  it('it renders correctly', () => {
+    const article = {
+      'title': 'New Article #3',
+      'image': '/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg'
```
Comment (krisquigley) [inline test/components/featuredArticle.test.js]: This requires a `slug` prop

File: test/components/userComments.test.js [hunk 17]
```diff
@@ -2,10 +2,25 @@ import UserComments from '../../components/UserComments'
 
 describe('<UserComments  />', () => {
   it('it renders correctly', () => {
+    const comments = [{
+      id: 'ID',
```
Comment (krisquigley) [inline test/components/userComments.test.js]: `id`, `articleId` aren't needed. But `createdAt` is

Comment (paulbrighton) [reply test/components/userComments.test.js]: I can remove articleId but if I remove id, the test fails

Comment (krisquigley) [reply test/components/userComments.test.js]: What's the error message? Are you using the ID for the key? If so, probably best to add it back to the query then.

Comment (paulbrighton) [reply test/components/userComments.test.js]: Warning: Each child in an array or iterator should have a unique "key" prop.%s%s See https://fb.me/react-warning-keys for more information.%s

**Pull request merged**